packages/coachmark/src/Coachmark.ts

```diff
@@ -13,32 +13,412 @@ governing permissions and limitations under the License.
+    primaryCTA?: string;
+
+    @property({ type: String, attribute: 'secondary-cta' })
+    secondaryCTA?: string;
+
+    constructor() {
+        super();
+    }
+
+    private interactOnVideo(): void {
+        if (!this.videoAsset) return;
+
+        if (this.canPlay) {
+            this.videoPlayPromise = this.videoAsset.play();
+            this.videoPlayPromise.catch((error) => {
+                console.error(error);
+            });
+        } else {
+            this.pauseVideo();
+        }
+    }
+
+    private pauseVideo(): void {
+        if (this.videoPlayPromise !== undefined) {
+            this.videoPlayPromise.then(() => {
+                this.videoAsset?.pause();
+            });
+        }
+    }
+    // render video and images
+    private renderMedia(): TemplateResult {
+        const isVideo = this.mediaType === MediaType.VIDEO;
+        const isImage = this.mediaType === MediaType.IMAGE;
+        this.hasAsset = isVideo || isImage;
+
+        if (!isVideo && !isImage) {
+            return html`
+                <slot name="asset"></slot>
+            `;
+        }
+
+        if (isVideo) {
+            return html`
+                <sp-asset class="asset">
+                    <video loop muted preload="auto">
+                        <source
+                            src="${ifDefined(this.source)}"
+                            type="${ifDefined(this.videoType)}"
+                        />
+                    </video>
+                </sp-asset>
+            `;
+        }
+
+        return html`
+            <sp-asset id="cover-photo">
+                <div class="image-wrapper">
+                    <img
+                        class="image"
+                        loading="lazy"
+                        slot="cover-photo"
+                        src="${ifDefined(this.source)}"
+                        alt="${ifDefined(this?.content?.imageAlt)}"
+                    />
+                </div>
+            </sp-asset>
+        `;
+    }
+    // method to render modifier
+    private renderModifier(
+        modifierKey: string,
+        type: 'modifier' | 'shortcut' = 'modifier'
+    ): TemplateResult {
+        return html`
+            <span type="${type}" class="keyboard-shortcut">${modifierKey}</span>
+        `;
+    }
+
+    private renderJoiner(): TemplateResult {
+        return html`
+            <span class="plus">&plus;</span>
+        `;
+    }
+    // render heading title and modifier
+    private renderHeader(): TemplateResult {
+        const hasModifier = this.modifierKeys && this.modifierKeys?.length > 0;
+        const hasShortcut = Boolean(this.shortcutKey);
+        const hasTitle = Boolean(this.content?.title);
+        if (!hasTitle && !hasModifier && !hasShortcut) {
+            return html`
+                <div class="title"><slot name="title"></slot></div>
+            `;
+        }
+        return html`
+            ${hasTitle
```

Comment:
Should never be optional. Title should always render.

Reply:
Ah, are you perhaps giving people the option if they don't use `content.title` to slot in their content?

Reply:
Yes the user can either put in the title as an attribute or as a slot.